include/frg/random.hpp

```diff
@@ -63,6 +64,43 @@ struct mt19937 {
 	int _ctr;
 };
 
+// Implementation based on https://github.com/imneme/pcg-c-basic
+// Copyright 2014 Melissa O'Neill <oneill@pcg-random.org>
+// Licensed under the Apache License, Version 2.0
+struct pcg_basic32 {
+	pcg_basic32(uint64_t seed, uint64_t seq = 1) {
+		this->seed(seed, seq);
+	}
+
+	void seed(uint64_t seed, uint64_t seq = 1) {
+		state = 0;
+		inc = (seq << 1) | 1;
+		operator()();
+		state += seed;
+		operator()();
+	}
+
+	uint32_t operator()() {
+		uint64_t oldstate = this->state;
+		this->state = oldstate * 6364136223846793005ULL + this->inc;
```

Comment:
Same here, why not suffix the private members with a `_` to indicate that they're private members.

Reply:
could do